### DISCUSSION TITLE
How to disable a field in create/edit form?

### DISCUSSION THREAD

**Discussion by hassyyy** [avo-hq/avo#1017] [Help] [answered]
I don't want certain fields in the resource to be changed by the user. Instead, the app should set the default value. For example, month & year field in a resource where the app should set the current month  & year automatically without user interaction

I tried using the following options but none of them work as expected

- readonly 
- hidden
- visible

All the above disables/hide the field from view but it doesn't work when the fields are required as the model throws error. On checking the logs, those field values are not submitted to the request at all

**Comment by adrianthedev** [comment]
Can you please send in a code sample and what didn't work?

**Comment by hassyyy** [comment]
**Resource:**
```ruby
class ContributionResource < Avo::BaseResource
  self.title = :id
  self.includes = []
  self.record_selector = false

  field :user, as: :belongs_to, readonly: true, required: true, name: 'Contributor', default: -> { context[:user].id }
  field :value, as: :number, required: true
  field :month, as: :text, readonly: true, required: true, default: -> { Date.today.strftime("%b") }
  field :year, as: :number, readonly: true, required: true, default: -> { Date.today.year }
  field :status, as: :badge, options: { info: ['submitted'], success: 'sent' }
  field :status, as: :status, hide_on: [:show, :index], required: true, default: 'submitted'
end
```
Fields are disabled with default values but I'm getting error on save
<img width="1159" alt="Error" src="https://user-images.githubusercontent.com/23002310/177326261-7433e1d9-3950-4ddf-b53e-3a517185d677.png">

Server Logs:
```shell
Started POST "/avo/resources/contributions" for 127.0.0.1 at 2022-07-05 17:53:00 +0530
Processing by Avo::ContributionsController#create as TURBO_STREAM
  Parameters: {"authenticity_token"=>"[FILTERED]", "contribution"=>{"user_id"=>"1", "value"=>"1213", "status"=>"submitted"}}
```

**Comment by hassyyy** [comment, marked as answer]
Is there a way to hide those fields from the user in forms and just set the default while saving the record?

**Reply by adrianthedev** [reply]
Use the `hidden` field.

https://docs.avohq.io/2.0/fields.html#hidden

**Reply by hassyyy** [reply]
It works, thanks 🙏

**Reply by hassyyy** [reply]
Hey @adrianthedev, I want the fields to be only hidden in forms. I still want the fields to be displayed in other views like index, show
How to achieve that?

**Reply by adrianthedev** [reply]
You add different fields for show, index and forms.

```ruby
  field :first_name, as: :text, hide_on: :forms # this will be displayed on Index and Show and hidden on Edit and New
  field :first_name, as: :hidden, only_on: :forms # this will act the opposite from the one above
```

**Reply by hassyyy** [reply]
Awesome 👍
